src/test/ui/type-alias-impl-trait/self-referential.stderr

```diff
@@ -1,10 +1,10 @@
-error[E0277]: can't compare `&i32` with `impl PartialEq<Opaque(DefId(0:6 ~ self_referential[5b7d]::Bar::{opaque#0}), [ReFree(DefId(0:7 ~ self_referential[5b7d]::bar), BrNamed(DefId(0:8 ~ self_referential[5b7d]::bar::'a), 'a)), ReEarlyBound(0, 'b)])> + Debug`
+error[E0277]: can't compare `&i32` with `impl PartialEq<Bar<'a, 'b>::{opaque#0}> + Debug`
```

Comment:
This exposed a deeper bug: the error should be about `<'b, 'a>`, but it talks about `<'a, 'b>`. Needs more investigation, hopefully just a diagnostic bug, as otherwise other things would be broken, too.

Reply:
Aha, found it. The diagnostic was trying to print the trait bounds of the opaque type, but TAITs can actually be mentioned by name/path, so I moved to doing that now. The printing infrastructure for printing trait bounds is very much tied to RPIT which knows exactly in which order its generic arguments are (namely in the order of its parent function). TAITs' generic arguments can be in any order since the user can specify them explicitly. At that point, printing the trait bounds becomes a fun exercise of substituting bounds.

Anyway, the new diagnostics are what we wanted (well I wanted) anyway, and from the looks of the not-working path printing impl, eddyb wanted it, too, and someone (probably me) broke that printing when we moved from `existential type` to `type Foo = impl Bar;`.